src/main/java/core/basesyntax/strategy/TransactionProcessor.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.Map;
+
+public class TransactionProcessor {
+    private Map<String, Integer> fruits;
+
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            String fruit = transaction.fruit();
+            int quantity = transaction.quantity();
+            switch (transaction.operation()) {
+                case BALANCE:
+                case SUPPLY, RETURN:
+                    fruits.put(fruit, fruits.getOrDefault(fruit, 0) + quantity);
+                    break;
+                case PURCHASE:
+                    fruits.put(fruit, fruits.getOrDefault(fruit, 0) - quantity);
+                    break;
+                default:
```

Comment:
Try to change for map implementation of strategy

Reply:
not fixed. Your new implementation simply takes single transaction instead of a list of transactions, which is wrong. Instead, the previous mentor meant that you should avoid using switch case and leverage a map of strategies. By the way, I don't see you implemented a strategy pattern. Why?